src/ripple/basics/Buffer.h

```diff
@@ -34,14 +34,49 @@ namespace ripple {
 class Buffer
 {
 private:
-    std::unique_ptr<
-        std::uint8_t[]> p_;
+    std::unique_ptr<std::uint8_t[]> p_;
     std::size_t size_ = 0;
 
 public:
     Buffer() = default;
-    Buffer (Buffer const&) = delete;
-    Buffer& operator= (Buffer const&) = delete;
+
+    /** Create an uninitialized buffer with the given size. */
+    explicit
+    Buffer (std::size_t size)
+        : p_ (size ? new std::uint8_t[size] : nullptr)
```

Comment:
General question: why construct `p_` with a `new` instead of `make_unique`? i.e.
```
: p_ (size ? std::make_unique<std::uint8_t[]>(size) : nullptr)
```
You get an extra `unique_ptr` which is then moved, but it does seem more C++14ish.